index.js

```diff
@@ -47,7 +59,22 @@ exports.register = function(server, options, next) {
                 request.response.source.context = {};
             }
 
-            request.response.source.context.notices = data.notices;
```

Comment:
there's enough logic sneaking in here that I'd be tempted to move it into a helper function that `server.ext('onPreResponse')` invokes.

`function assignNoticesToContext () ???`